Autopilot/PathManager/Src/pathStateClasses.cpp

```diff
@@ -120,4 +164,287 @@ pathManagerState& fatalFailureMode::getInstance()
     return singleton;
 }
 
+/****************************************************************************************************
+
+LANDING STATE FUNCTIONS
+
+****************************************************************************************************/
+
+void landingTransitionStage::execute(pathManager* pathMgr)
+{
+    if(!pathMgr -> madeLandingPoints)
+    {
+        //requires data structure that dhruv wants to use 
+        landingPath = WaypointManager(0.0,0.0);
+        path = LandingManager::createSlopeWaypoints(getFromTelemetry::telemetryInput, sensorFusion::sensorInput.altitude);
```

Comment:
if you are creating a new waypoint manager you can actually make a constructor for LandingManager. So getting rid of the inheritance should not be too complex

Reply:
There doesn't need to be a constructor, landing and takeoff classes are just a class of functions.